docs/content/en/docs/design/config.md

```diff
@@ -25,6 +21,43 @@ read the configuration file from the current directory.
 
 You can [learn more]({{< relref "/docs/references/yaml" >}}) about the syntax of `skaffold.yaml`.
 
+Skaffold normally expects to find the configuration file as
+`skaffold.yaml` in the current directory, but the location can be
+overridden with the `--filename` flag.
+
+### File resolution
```

Comment:
Can you add a `File resolution` section down below or mention here itself about configurations resolved as dependencies. The paths in those files are resolved relative to the directory containing the imported file.  So it's _recommended_ to have the `skaffold.yaml` file live in the root of the project and not in a nested directory.